lib/mongoid/matcher.rb

```diff
@@ -50,8 +47,9 @@ module Matcher
         current.each do |doc|
           case doc
           when Hash
-            if doc.key?(field)
-              new << doc[field]
+            v = indifferent_hash_fetch(doc, field)
+            if !v.nil?
+              new << v
             end
           when Array
             if (index = field.to_i).to_s == field
```

Comment:
Re-looking at this, I think this line needs to be:
```suggestion
            if (index = field.to_i).to_s == field.to_s
```

Reply:
Why do I need to call `field.to_s`? Won't `field` always be a string because it's one item within the array result of a `String#split` call?

Reply:
My bad there, you're totally right